lib/statsd.rb

```diff
@@ -476,9 +476,11 @@ def send_to_socket(message)
       # send(2) is atomic, however, in stream cases (TCP) the socket is left
       # in an inconsistent state if a partial message is written. If that case
       # occurs, the socket is closed down and we retry on a new socket.
-      n = socket.write(message)
+      l = @protocol == :tcp ? message.length + 1 : message.length
+      br = @protocol == :tcp ? "\n" : ""
+      n = socket.write(message + br)
 
-      if n == message.length
+      if n == l
```

Comment:
in githubs chosen font, i can't tell if this is a 1 or an l, so it's a readability problem. Could you use len or length?

I think it might be just as easy to write this:

```ruby
message = @protocol == :tcp ? message + "\n" : message
n = socket.write(message)
if n == message.length
```

Reply:
Thanks @raggi - you're right, that looks better! I've amended the commit.